core/tx_pool.go

```diff
@@ -982,6 +982,9 @@ func (pool *TxPool) journalTx(from common.InternalAddress, tx *types.Transaction
 //
 // Note, this method assumes the pool lock is held!
 func (pool *TxPool) promoteTx(addr common.InternalAddress, hash common.Hash, tx *types.Transaction) bool {
+	if types.IsCoinBaseTx(tx) {
```

Comment:
I *think* this check is unnecessary because what we should be checking is if it's an ETX. But that check should happen in addTx, basically upon first receiving the tx, otherwise the node will crash in types.Sender(tx)

Reply:
removed, but we need to make sure we only add Quai txs back into the pool

Reply:
I made an issue for it